includes/orders/functions/orders.php

```diff
@@ -949,13 +949,17 @@ function edd_build_order( $order_data = array() ) {
 			// Store order item fees as adjustments.
 			if ( isset( $item['fees'] ) && ! empty( $item['fees'] ) ) {
 				foreach ( $item['fees'] as $fee_id => $fee ) {
+					$tax = ( isset( $fee['no_tax'] ) && false === $fee['no_tax'] && ! empty( $tax_rate->amount ) ) || ( $fee['amount'] < 0 && ! empty( $tax_rate->amount ) )
+						? floatval( floatval( $fee['amount'] ) - ( floatval( $fee['amount'] ) / ( 1 + $tax_rate->amount ) ) )
+						: 0.00;
```

Comment:
I am glad you were ultimately able to find what the heck was going on with this.

If I had to guess, it looks like this approach was nicked from the migrator. (It was unruly then, and me thinks it's still quite a handful to work out what's happening.)

----

Each comparison on the side of the `||` checks for:

```php
! empty( $tax_rate->amount )
```
...so that could happen first with an `&&` and then get removed from both sides.

Statements like this...
```php
isset( $fee['no_tax'] ) && false === $fee['no_tax']
```
...could be simplified down to:

```php
empty( $fee['no_tax'] );
```

(I suspect that strict type checking against `'no_tax'` was a Stylistic choice by Sunny, but I'm not convinced it's worth the additional code complexity on these lines specifically, especially considering that the `'no_tax'` logic-gate already introduces a double-negative.)

Perhaps this is an opportunity to introduce a public static method in the `EDD_Fees` class to centralize these repeated calls, along with some global function akin to `edd_calculate_tax()` but that knowingly does not calculate tax for the `floatval( floatval( ... ) )` business.

----

👍 again for figuring this out.

Reply:
I can't claim to have figured anything out--just when I was documenting all uses of `edd_add_order_adjustment`, I discovered that this particular bit of code was missing the `tax` for fees, although all other fees have the potential tax added. I did pull the tax calculation logic from the data migrator, since it was already in place for that, but I agree that it's a bit convoluted. Would be happy to update it here and the other places where it's done this way.

Making this consistent was just something I felt I could go ahead and resolve while we are still thinking about the other elements of this data table.